08_homework_store_cart/script.js

```diff
@@ -0,0 +1,164 @@
+        description: "It is a retro standard motorcycle marketed by Kawasaki for model years 1999-2007" +
+                     "In 1999,Kawasaki introduced the W650, resembling British motorcycles of the early 1960s, notably the Triumph Bonneville."
+    },
+    {
+        brand: "Triumph",
+        model: "Thruxton",
+        engineCapacity: 865,
+        colour: "black",
+        imgSrc: 'images/triumph.jpg',
+        price: 6000,
+        currency: "Pounds",
+        condition: {
+            year: 2008,
+            mileage: 8000,
+            mileageUnit: "mi"
+        },
+        inStock: false,
+        description: "The Triumph Thruxton is a series of British motorcycles with parallel-twin engines and sports styling." +
+                     " The name Thruxton was first applied to a handbuilt machine for endurance racing in the mid 1960s, " +
+                     "and later revived in the 2000s. Production 2004 - 2016"
+    }
+];
+let inCart = [];
+function bikeListDisplay() {
+    for (let i=0; i < bikes.length; i++) {
+        bikeCard = document.createElement("div");
+        bikeCard.setAttribute("class", "card");
+        bikeCard.setAttribute("id", i);
+
+        const main = document.getElementById('main');
+        bikeCard.innerHTML = ` 
+          <h2>${bikes[i].brand} ${bikes[i].model}</h2>
+          <img class="bikePic" src="${bikes[i].imgSrc}" />
+          <li>Price: ${bikes[i].price} ${bikes[i].currency}</li>
+          <li>Year: ${bikes[i].condition.year}</li>
+          <li>Mileage: ${bikes[i].condition.mileage} ${bikes[i].condition.mileageUnit}</li>
+          <li>Colour: ${bikes[i].colour}</li>
+          <li>Engine: ${bikes[i].engineCapacity} cc</li>
+          <p>${bikes[i].description}</p>
+        `;
+
+        const buttonElement = document.createElement('button');
+        buttonElement.innerText = 'Add to cart';
+        buttonElement.setAttribute("class", "button");
+        buttonElement.setAttribute("id", "addToCart");
+        buttonElement.addEventListener('click', function() {
+            inCart.push(bikes[i]);
+            buttonElement.setAttribute("disabled", "true");
+            buttonElement.setAttribute("class", "buttonDisabled");
+        });
+        bikeCard.appendChild(buttonElement);
+        main.appendChild(bikeCard);
+    }
+};
+bikeListDisplay();
+function showCartButton() {
+    document.querySelectorAll('.addToCart').forEach(item => {
+        item.addEventListener('click', event => {
+            addButton()
+        })
+    })
+};
+
+function addButton() {
+    if (inCart.length >0) {
+        console.log('Array not empty');
+    }
+};
+
+showCart.addEventListener('click',cartListDisplay);
+function cartListDisplay() {
+    let totalSumm=0;
+    for (let i=0; i < inCart.length; i++) {
+        totalSumm = totalSumm + inCart[i].price;
+        bikeCard = document.createElement("div");
+        bikeCard.setAttribute("class", "card");
+        bikeCard.setAttribute("id", i);
```

Comment:
you don't need to set id, because you're not using it. The same rule applies to elements and for code: if you don't use something don't put it on the page. Then your code would be smaller and have less places to fail.

Reply:
Fixed.